Libraries/LibURL/URL.cpp

```diff
@@ -428,7 +428,7 @@ bool code_point_is_in_percent_encode_set(u32 code_point, PercentEncodeSet set)
     case PercentEncodeSet::SpecialQuery:
         return code_point_is_in_percent_encode_set(code_point, PercentEncodeSet::Query) || code_point == '\'';
     case PercentEncodeSet::Path:
-        return code_point_is_in_percent_encode_set(code_point, PercentEncodeSet::Query) || "?`{}"sv.contains(static_cast<char>(code_point));
+        return code_point_is_in_percent_encode_set(code_point, PercentEncodeSet::Query) || "?^`{}"sv.contains(static_cast<char>(code_point));
     case PercentEncodeSet::Userinfo:
         return code_point_is_in_percent_encode_set(code_point, PercentEncodeSet::Path) || "/:;=@[\\]^|"sv.contains(static_cast<char>(code_point));
```

Comment:
Like the spec, we can remove `^` here as it is now covered by the path encoding set.